docs/editor/intellisense.md

```diff
@@ -0,0 +1,162 @@
+
+![IntelliSense demo](images/intellisense/intellisense.gif)
+
+## Where is IntelliSense for my language? 
+
+IntelliSense is provided in VS Code for JavaScript, TypeScript, JSON, HTML, CSS, Less, and Sass out of the box. Other languages have word completions and can be configured to have IntelliSense by installing a language extension. 
+
+Below are the most popular language extensions in the [Marketplace](https://marketplace.visualstudio.com/vscode). Click on an extension tile below to read the description and reviews to decide which extension is best for you. 
+
+> **Tip:** For configuring and troubleshooting JavaScript IntelliSense see [here](/docs/languages/javascript#configuring-intellisense).
+
+<div class="marketplace-extensions-languages"></div>
+
+## IntelliSense Features
+
+If a language service knows possible completions, the IntelliSense suggestions will pop up as you type. If you continue typing characters, the list of members (variables, methods, etc.) is filtered to include only members containing your typed characters. Pressing `kbstyle(Tab)` or `kbstyle(Enter)` will insert the selected member. 
+
+You can trigger IntelliSense in any editor window by typing `kb(editor.action.triggerSuggest)` or by typing a trigger character (such as `.` in JavaScript). 
+
+![intellisense in package json](images/intellisense/intellisense_packagejson.gif)
+
+> **Tip:** The suggestions filtering supports CamelCase so you can type the upper case letters of a method name to limit the suggestions. For example, "cra" will quickly bring up "createApplication".
+
+> **Tip:** You can turn off IntelliSense while you type. See [Customize IntelliSense below](#_customize_intellisense).
+
+In addition, you can see **quick info** for each method (as provided by the language service). 
+
+![quick info](images/intellisense/quick_outline.png)
+
+After choosing a method you are provided with **parameter info**. 
+
+![parameter info](images/intellisense/paramater_info.png)
+
+> **Tip:** When applicable, a language service will surface the underlying types in the quick info and method signatures. In the image above, you can see serveral `any` types. Because JavaScript is dynamic and doesn't need or enforce types, `any` suggests that the variable can be of any type. 
+
+
+## Suggestion Priority
+
+The order and the icon to the left of the word show the IntelliSense suggestion's priority.  
+
+> **Higher suggestion priority = higher order and a more detailed icon**
+
+This image shows the result of triggering IntelliSense on `express` a popular [Node.js module](https://expressjs.com/). 
+
+![image showing intellisense icons](images/intellisense/intellisense_icons.png)
+
+Methods, variables, and objects tend to be high priority. Simple word completion, tend to be lower priority. When the priority level is equivalent, ordering is done alphabetically. 
```

Comment:
The screenshot is misleading and doesn't illustrate the point you make below with regard to the sort order. The screen shot it doesn't show any word completions. All the completions come from the JavaScript language service. Since JavaScript is a loosely typed language it shows the inferred completions first and then it adds the global symbols.
From the doc: "IntelliSense offers both inferred proposals and the global identifiers of the project. The inferred symbols are presented first, followed by the global identifiers (with the document icon), as you can see in the image below."